niceml/utilities/readwritelock.py

```diff
@@ -136,15 +143,19 @@ def acquire(self):
             return
         with open_location(self.path_config) as (cur_fs, root_path):
             start_time = time.monotonic()
+            retry_time = 0
             while True:
                 write_lock_path = join_fs_path(cur_fs, root_path, self.write_lock_name)
+                cur_fs.mkdirs(root_path, exist_ok=True)
                 if is_lock_file_acquirable(write_lock_path, cur_fs):
                     break
                 if time.monotonic() - start_time > self.timeout:
                     raise TimeoutError(
                         f"Timeout while acquiring write lock {write_lock_path}"
                     )
+                logging.info(f"waiting for read lock release {retry_time} seconds")
                 time.sleep(self.retry_time)
```

Comment:
`retry_time` is a bit scrambling at this point. For the local variable, it is ok. I would suggest something like `self.retry_await_time` for the class variable if this is what is meant.

Reply:
implemented as suggested